frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -188,27 +204,30 @@ impl<T: Config> SignedSubmissions<T> {
 
 	/// Perform three operations:
 	///
-	/// - Remove a submission (identified by score)
-	/// - Insert a new submission (identified by score and insertion index)
-	/// - Return the submission which was removed.
-	///
-	/// Note: in the case that `weakest_score` is not present in `self.indices`, this will return
-	/// `None` without inserting the new submission and without further notice.
+	/// - Remove the solution at the given position of `self.indices`.
+	/// - Insert a new submission (identified by score and insertion index), if provided.
+	/// - Return the submission which was removed, if any.
 	///
-	/// Note: this does not enforce any ordering relation between the submission removed and that
-	/// inserted.
+	/// The call site must ensure that `remove_pos` is a valid index. If otherwise, `None` is
+	/// silently returned.
 	///
 	/// Note: this doesn't insert into `insertion_overlay`, the optional new insertion must be
-	/// inserted into  `insertion_overlay` to keep the variable `self` in a valid state.
+	/// inserted into `insertion_overlay` to keep the variable `self` in a valid state.
 	fn swap_out_submission(
 		&mut self,
-		remove_score: ElectionScore,
-		insert: Option<(ElectionScore, u32)>,
+		remove_pos: usize,
+		insert: Option<(ElectionScore, T::BlockNumber, u32)>,
 	) -> Option<SignedSubmissionOf<T>> {
-		let remove_index = self.indices.remove(&remove_score)?;
-		if let Some((insert_score, insert_idx)) = insert {
+		if remove_pos >= self.indices.len() {
+			return None
+		}
+
+		// safe: index was just checked in the line above.
+		let (_, _, remove_index) = self.indices.remove(remove_pos);
```

Comment:
The complexity this remove incurs is `O(n)` for `indices` having `n` elements and should therefore be benchmarked with `remove_pos = 0` for worst case performance.

Reply:
https://github.com/paritytech/substrate/pull/12237/commits/10a8a2c56c4b2b66cc571e53523cfdb1c5892760 should be a better benchmark in general.